connexion/middleware/security.py

```diff
@@ -81,30 +82,36 @@ def __init__(
 
     def add_auth_on_not_found(self):
         """Register a default SecurityOperation for routes that are not found."""
-        default_operation = self.make_operation()
+        default_operation = self.make_operation(self.specification)
         self.operations = defaultdict(lambda: default_operation)
 
     def add_paths(self):
         paths = self.specification.get('paths', {})
         for path, methods in paths.items():
-            for method, operation in methods.items():
+            for method in methods:
                 if method not in METHODS:
                     continue
-                operation_id = operation.get('operationId')
-                if operation_id:
-                    self.operations[operation_id] = self.make_operation(operation)
-
-    def make_operation(self, operation_spec: dict = None):
-        security = self.app_security
-        if operation_spec:
-            security = operation_spec.get('security', self.app_security)
-
-        return SecurityOperation(
-            self.security_handler_factory,
-            security=security,
-            security_schemes=self.specification.security_definitions
+                try:
+                    self.add_operation(path, method)
+                except ResolverError:
+                    # ResolverErrors are handled in routing middleware
```

Comment:
Why don't we propagate the error in this case?

If the resolving fails, it is normally already raised in the routing middleware, or am I mistaken?
But if it is already raised there, then we don't need to account for it here?

So, not sure what the flow is here :)

Reply:
Resolution is done at start up. So 
- if the `ResolverError` is not handled in the routing middleware, start up will fail and this code will indeed never be reached.
- If the `ResolverError` is handled in the routing middleware, it will still be raised here during start-up, but a request will never reach the operation since it will already be handled by the `resolver_error_handler` in the routing middleware.

I can add some more explanation to the comment, but open to suggestions on how to handle it more clearly in code.